File: 2837-minimum-operations-to-make-the-integer-zero/2837-minimum-operations-to-make-the-integer-zero.py
class Solution:
    def makeTheIntegerZero(self, num1: int, num2: int) -> int:
        res = 1
        num = num1
        while True:
            num -= num2
            if num < res:
                return -1
            elif res >= num.bit_count():
                return res
            res += 1

File: 2837-minimum-operations-to-make-the-integer-zero/2837-minimum-operations-to-make-the-integer-zero.cpp
class Solution {
public:
    int makeTheIntegerZero(int num1, int num2) {
        int res = 1;
        long long n = num1;
        while(1)
        {
            n -= num2;
            if (n < res)
            {
                return -1;
            }
            else if(res >= __builtin_popcountll(n))
            {
                return res;
            }
            res++;
        }
        return res;
    }
};